storage/queue.go

```diff
@@ -82,6 +82,24 @@ func (p PeekMessagesParameters) getParameters() url.Values {
 	return out
 }
 
+// UpdateMessageParameters is the set of options can be specified for Update Messsage
+// operation. A zero struct does not use any preferences for the request.
+type UpdateMessageParameters struct {
+	PopReceipt        string
+	VisibilityTimeout int
+}
+
+func (p UpdateMessageParameters) getParameters() url.Values {
+	out := url.Values{}
+	if len(p.PopReceipt) != 0 {
+		out.Set("popreceipt", p.PopReceipt)
+	}
+	if p.VisibilityTimeout != 0 {
```

Comment:
@amanohar I don't have a strong preference. You're right about `PutMessageParameters`. I guess while writing that struct I didn't think of `0` being a valid value people can specify... We can go with this code for now I guess. If somebody tries to specify it as `0`, the value will not be sent in the request though. So we'd have to take a breaking change by making these fields *int, but for now I guess it's fine.